source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ADCResolver.m

```diff
@@ -32,6 +32,30 @@ - (UIImageView *)resolveImageViewResource:(NSURL *)url {
     return imageView;
 }
 
+- (UIImageView *)resolveBackgroundImageViewResource:(NSURL *)url hasStretch:(BOOL)hasStretch {
```

Comment:
is this function added for other renderers?
can "imageView.backgroundColor = [UIColor colorWithPatternImage:image];" be part of the renderer, so user wouldn't make mistakes?

Reply:
Nope. This is only in the ios renderer, unfortunately. The problem I'm running into is that `resolveImageViewResource()` returns an imageView. To do the background image properly, the image has to be attached to the imageView in a specific way based on the backgroundImageMode. If `resolveImageResource()` was used, I think this would be doable because I could just attach the image to the imageView properly in `ACRRenderer`, `ColumnRenderer`, `ContainerRenderer`. Suggestions? Could I remove `resolveImageViewResource()` and make `resolveImageResource()` not-optional?

---
In reply to: [256536421](https://github.com/Microsoft/AdaptiveCards/pull/2388#discussion_r256536421) [](ancestors = 256536421)